### ISSUE TITLE
kra healthcheck issue

### ISSUE THREAD

**Issue by RobVerduijn** [freeipa/freeipa-healthcheck#212]
Hello,

Today I've updated ipa to 4.9.2, but now the healthcheck gives a warning.
googleing did not yield any answers,
Anybody here who knows where to start ?

```
ipa-healthcheck --failures-only
Internal error testing KRA clone. 'NoneType' object has no attribute 'config'
[
  {
    "source": "pki.server.healthcheck.clones.connectivity_and_data",
    "check": "ClonesConnectivyAndDataCheck",
    "result": "ERROR",
    "uuid": "ae1a813c-9f6a-4792-8b03-21bf012dfa08",
    "when": "20210526160521Z",
    "duration": "0.128475",
    "kw": {
      "status": "ERROR:  pki-tomcat : Internal error testing KRA clone. Host: freeipa02.tjako.thuis Port: 443"
    }
  }
]
```

**Comment by rcritten** [comment]
@edewata this seems to be an issue in the pki healthcheck code. Do you want me to clone this ticket?

**Comment by edewata** [comment]
@rcritten Sure. Please provide the PKI version number too. So is this a regression? Thanks.

**Comment by RobVerduijn** [comment]
```
pki --version
PKI Command-Line Interface 10.10.5-2.module_el8.5.0+737+ee953a1e
```

**Comment by flo-renaud** [comment]
Hi @RobVerduijn 
The issue looks familiar, please see https://pagure.io/freeipa/issue/8582.

@edewata there is additional info in the above ticket, with the user entry mapped to the client cert used for authentication. I didn’t find any pki ticket but I’m sure we already discussed the issue.

**Comment by RobVerduijn** [comment]
> Hi @RobVerduijn
> The issue looks familiar, please see https://pagure.io/freeipa/issue/8582.
> 
> @edewata there is additional info in the above ticket, with the user entry mapped to the client cert used for authentication. I didn’t find any pki ticket but I’m sure we already discussed the issue.

Hi,
I'm rather busy at the moment, I'll dig in to it at the end of the week.

Rob

**Comment by rcritten** [comment]
Someone else just saw this on freeipa-users. Could this mean it is a leftover in the securitydomain?

**Comment by rcritten** [comment]
Cloned to PKI https://github.com/dogtagpki/pki/issues/3608

**Comment by amessina** [comment]
Perhaps related, running ipa-healthcheck on my first master (ipa484a), I get warnings related to my second master (ipa484b).  The link-local below is that of the first master:
```
Internal error testing KRA clone. KRA clone problem detected  Host: ipa484b.messinet.com Port: 443
Invalid IP address fe80::ccfd:21ff:feaf:7417 for ipa484a.messinet.com.: cannot use link-local IP address fe80::ccfd:21ff:feaf:7417
```
```
pki --version
PKI Command-Line Interface 11.0.0-1.fc35
```

**Comment by rcritten** [comment]
The root cause is that IPA wasn't cleaning up the PKI security domain when removing servers, https://bugzilla.redhat.com/show_bug.cgi?id=1740702 
Assuming this host is no longer available, or was re-installed without the KRA service, it can be removed by following the instructions in the BZ.
The IP message comes from IPA and can be ignored. It is a side-effect of ensuring that the known IP addresses are valid in DNS. I may try to suppress it but it's only a message on output, not an error recorded by healthcheck.

**Comment by amessina** [comment]
In my case the KRA's should exist.  They were created at the same time I created the installations of the two masters.

**Comment by flo-renaud** [comment]
> In my case the KRA's should exist. They were created at the same time I created the installations of the two masters.

@amessina in this case you're rather hitting issue https://pagure.io/freeipa/issue/8582 / https://github.com/dogtagpki/pki/issues/3608

**Comment by amessina** [comment]
Thank you @flo-renaud, I am "subscribed" to https://pagure.io/freeipa/issue/8582 & https://github.com/dogtagpki/pki/issues/3608 awaiting activity in either.

**Comment by rcritten** [comment]
Looks like the PKI issue was fixed in pki 11.1.0 but not backported to 10.x

**Comment by rcritten** [comment]
The reported error was fixed in the PKI v10.12 branch in 7d62105c676fc79e0c32766c41cd034655a524ff

The related IPA ticket added a missing proxy for the KRA status page in https://pagure.io/freeipa/issue/8582 in v4.9.9.

So I guess this is resolved with the right package versions.
